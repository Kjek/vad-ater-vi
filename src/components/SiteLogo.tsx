import { api } from '@util/api';
import Image from 'next/image';

const SiteLogo = () => {
  return (
    <>
      <Image
        src='/site-image.png'
        alt='Lunch Image'
        width={40}
        height={40}
        className='h-fit'
      />
    </>
  );
};

export default SiteLogo;
