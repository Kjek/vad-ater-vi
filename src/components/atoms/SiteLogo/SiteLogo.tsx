import { cn } from '@util/cn';
import Image from 'next/image';
import { type ComponentPropsWithoutRef } from 'react';

const SiteLogo = ({ className, ...props }: ComponentPropsWithoutRef<'img'>) => {
  return (
    <>
      <Image
        {...props}
        src='/site-image.png'
        alt='Lunch Image'
        width={40}
        height={40}
        className={cn('h-fit', className)}
      />
    </>
  );
};

export default SiteLogo;
