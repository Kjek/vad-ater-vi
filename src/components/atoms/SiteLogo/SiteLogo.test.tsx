import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import SiteLogo from './SiteLogo';

describe(SiteLogo, () => {
  it('renders correctly', () => {
    const { getByTestId } = render(<SiteLogo data-testid={'site.logo'} />);
    const siteLogo = getByTestId('site.logo');
    expect(siteLogo).toBeInTheDocument();
    expect(siteLogo).toBeVisible();
  });

  it('matches snapshot', () => {
    const defaultRender = render(<SiteLogo />);
    expect(defaultRender).toMatchSnapshot();
  });
});
