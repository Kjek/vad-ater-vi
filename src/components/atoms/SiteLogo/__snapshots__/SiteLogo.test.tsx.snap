// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SiteLogo > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Lunch Image"
        class="h-fit"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fsite-image.png&w=96&q=75"
        srcset="/_next/image?url=%2Fsite-image.png&w=48&q=75 1x, /_next/image?url=%2Fsite-image.png&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Lunch Image"
      class="h-fit"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fsite-image.png&w=96&q=75"
      srcset="/_next/image?url=%2Fsite-image.png&w=48&q=75 1x, /_next/image?url=%2Fsite-image.png&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
